### PR TITLE
Add conditionals to check the presence of fields in json response

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [ ] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [x] Bug fix or refactor (change which fixes an issue)
- [ ] New feature (change which adds functionality)

## Summary of the change.

- added error response when attempting to create shopping list without items
- removed the `name` and `productId` from the `item` json
- added conditionals to check the presence of attributes in Kroger API response for `items` resource 

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: trackbasket_be/models/item.py [hunk 1]
```diff
@@ -17,11 +17,9 @@ class Item(BaseModel, db.Model):
   
   def json(self):
     return {
-      'name': self.name,
```
Comment (adumortier) [inline trackbasket_be/models/item.py:20]: remove fields that are not needed by the front end

File: trackbasket_be/models/krogerservice.py [hunk 2]
```diff
@@ -66,27 +66,33 @@ def item_search(cls, term, at_risk_user_id):
     nearest_store = user.store[0]
     nearest_store_id = nearest_store.location_id
     parameters = {'filter.term': term, 'filter.locationId': nearest_store_id}
+    if term == '':
+      return 'error'
```
Comment (adumortier) [inline trackbasket_be/models/krogerservice.py:70]: checks that a search term was passed in the request

File: trackbasket_be/models/krogerservice.py [hunk 3]
```diff
@@ -66,27 +66,33 @@ def item_search(cls, term, at_risk_user_id):
...
+      json_items = response.json()['data']
+      items = []
+      for item in json_items:
+        data = {}
+        data['description'] = item['description'] if 'description' in item.keys() else ''
+        data['upc'] = item['upc'] if 'upc' in item.keys() else ''
+        if ('aisleLocations' in item.keys()) and (item['aisleLocations'] != []):
+          data['aisle_number'] = int(item['aisleLocations'][0]['number'])
+        if ('price' in item['items'][0].keys()):
+          data['unit_price'] = item['items'][0]['price']['regular']
```
Comment (adumortier) [inline trackbasket_be/models/krogerservice.py:85]: checks the presence of keys in the `response.json()` hash to handle gracefully situations where some attributes are missing from the response (try to search for the item 'oragne' for example)

File: trackbasket_be/resources/shopping_list.py [hunk 4]
```diff
@@ -16,6 +16,8 @@ def post(self,id):
     if at_risk_user is None:
       return {'data': { 'id': 'shoppinglist', 'attributes': {'error': "At risk user #{} not found".format(id)} } }, 400
     else:
+      if request_data["items"] == []:
+        return {'data': { 'id': 'shoppinglist', 'attributes': {'error': "Shopping List can not be empty"} } }, 400
```
Comment (adumortier) [inline trackbasket_be/resources/shopping_list.py:20]: returns error when trying to create an empty `shopping list`

File: trackbasket_be/models/krogerservice.py (hunk 3 above)
Comment (mronauli) [reply trackbasket_be/models/krogerservice.py:85]: good call on this edge case

File: trackbasket_be/models/item.py (hunk 1 above)
Comment (mronauli) [inline trackbasket_be/models/item.py:20]: developer empathy ftw!

**Pull request merged**